src/components/GameTile.vue

```diff
@@ -0,0 +1,350 @@
+<script setup>
+import { reactive, defineProps, computed } from "vue";
+import CustomButton from "@/components/CustomButton.vue";
+
+const props = defineProps({
+  tile: {
+    type: Object,
+    required: true,
+    default: () => null,
+  },
+  currency: {
+    type: String,
+    default: "USD",
+    required: true,
+  },
+});
+
+const platformIconStyle = computed(() => {
+  if (props.tile?.platform) {
+    try {
+      const iconUrl = require(`@/assets/GameTile/${props.tile.platform}-logo.png`);
+      return { "--platform-icon": `url(${iconUrl})` };
+    } catch (e) {
+      console.error(`Cannot load icon for platform: ${props.tile.platform}`);
+      return {};
+    }
+  }
+  return {};
+});
+
+function truncateDescription(text, maxLength) {
+  if (text.length <= maxLength) return text;
+  let truncated = text.slice(0, maxLength);
+  const lastSpace = truncated.lastIndexOf(" ");
+  if (lastSpace > 0) {
+    truncated = truncated.slice(0, lastSpace);
+  }
+  return truncated + "...";
+}
+
+const addToCartButton = reactive({
+  iconSource: require(`@/assets/GameTile/shopping-cart_white.png`),
+  iconAlternativeText: "shopping-cart_white.png",
+  label: "ADD TO CART",
+  disabled: false,
+});
+
+function isDiscount() {
+  const discountInPercentage = props.tile?.discountInPercentage ?? 0;
+  return discountInPercentage > 0;
+}
```

Comment:
```suggestion
const hasDiscount = computed(
    () => (props?.tile?.discountInPercentage ?? 0) > 0
);
```

In Vue, functions that calculate values based on reactive data (like props) are perfect candidates for computed properties. They are cached based on their reactive dependencies, meaning they only re-run when a dependency changes. Your isDiscount and discountedPrice functions should be converted to computed properties. Read more [here](https://vuejs.org/guide/essentials/computed).